app/src/main/java/com/github/se/eventradar/ui/login/SignUpScreen.kt

```diff
@@ -0,0 +1,469 @@
+                }
+            }
+            Row(
+                modifier = Modifier
+                    .constrainAs(ref = usernameField) {
+                        top.linkTo(
+                            profilePicture.bottom,
+                            margin = 16.dp
+                        )
+                        centerHorizontallyTo(parent)
+                    },
+                horizontalArrangement = Arrangement.Center,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                OutlinedTextField(
+                    value = username.value,
+                    onValueChange = { username.value = it },
+                    label = { Text("Username") },
+                    modifier = Modifier.width(320.dp),
+                    colors = TextFieldDefaults.outlinedTextFieldColors(
+                        focusedBorderColor = MaterialTheme.colorScheme.primary,
+                        unfocusedBorderColor = MaterialTheme.colorScheme.primary
+                    ),
+                    shape = RoundedCornerShape(12.dp),
+                    leadingIcon = {
+                        Text(
+                            "@",
+                            Modifier.padding(start = 12.dp),
+                            MaterialTheme.colorScheme.primary
+                        )
+                    },
+                    isError = username.value.isEmpty(),
+
+                    )
+            }
+            Row(
+                modifier = Modifier.constrainAs(ref = nameField) {
+                    top.linkTo(
+                        usernameField.bottom,
+                        margin = 16.dp
+                    )
+                    centerHorizontallyTo(parent)
+                },
+                horizontalArrangement = Arrangement.Center,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                OutlinedTextField(
+                    value = name.value,
+                    onValueChange = { name.value = it },
+                    label = { Text("Name") },
+                    modifier = Modifier.width(320.dp),
+                    colors = TextFieldDefaults.outlinedTextFieldColors(
+                        focusedBorderColor = MaterialTheme.colorScheme.primary,
+                        unfocusedBorderColor = MaterialTheme.colorScheme.primary
+                    ),
+                    shape = RoundedCornerShape(12.dp),
+                    isError = name.value.isEmpty()
+                )
+            }
+            Row(
+                modifier = Modifier.constrainAs(ref = surnameField) {
+                    top.linkTo(
+                        nameField.bottom,
+                        margin = 16.dp
+                    )
+                    centerHorizontallyTo(parent)
+                },
+                horizontalArrangement = Arrangement.Center,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
```

Comment:
nit: Just a suggestion for code modularization: could create a separate compose function e.g. fun UserInputTextField for each one of these. But can do it later due to Milestone 1 deadline.

Reply:
Yep! Was thinking while writing all the visual elements of this page that there was a lot of overlap. Will focus on this next sprint to make the overall code a lot more optimised.